Katello/katello
# Fixes #35263 - Index ModuleStreamErratumPackages properly

#### What are the changes introduced in this pull request?
Indexes ModuleStreamErratumPackages for all duplicate errata. Also includes a migration to reindex related library repositories.

#### Considerations taken when implementing this change?
It can be hard to tell which repositories exactly are affected by bug, so I reindex all repositories that have both duplicated errata and module streams.

#### What are the testing steps for this pull request?
1) Sync AppStream
2) Find the module stream with version 8050020210712211742, name httpd, stream 2.4
3) Find the erratum with ID RHSA-2021:4257
4) Duplicate the erratum with a dummy pulp_id
5) Move the all related ErratumPackages to the duplicate:
```ruby
erratum_packages = ::Katello::ErratumPackage.find(Katello::ModuleStreamErratumPackage.joins(:erratum_package => {:erratum => :repository_errata}).where("#{::Katello::RepositoryErratum.table_name}.repository_id" => appstream_repo_id).where("#{Katello::ModuleStreamErratumPackage.table_name}.module_stream_id = module_stream_id_from_above").pluck(:erratum_package_id))
erratum_packages.each {|ep| ep.update(erratum_id: duplicate_erratum.id)}
```
6) Check that appstream_repo.module_streams_without_errata returns the module stream mentioned above
7) Run appstream_repo.index_content
8) Note that appstream_repo.module_streams_without_errata still returns the module stream mentioned above
9) Patch in my changes
10) Run index_content again (in the PR: run the migration)
11) See that appstream_repo.module_streams_without_errata no longer returns the module stream mentioned above
